cactus/tests/test_plugins.py

```diff
@@ -71,4 +76,143 @@ def test_call(self):
 
         #postBuild
         self.assertEqual(1, len(plugin.postBuild.calls))
-        self.assertEqual((self.site,), plugin.postBuild.calls[0]['args'])
\ No newline at end of file
+        self.assertEqual((self.site,), plugin.postBuild.calls[0]['args'])
+
+
+class TestPagecontextPlugin(unittest.TestCase):
+
+    def test_jekyll_style(self):
+        testdata = [
+            (['---',  # fenced
+              'Author : Me',
+              'Content: Awesome',
+              '---',
+              'Here comes the content'],
+             {'Author': 'Me', 'Content': 'Awesome'},
+             ['Here comes the content']),
+
+            (['Author : Me',  # no start of metadata
+              '---',
+              'Here comes the content'],
+             {},
+             ['Author : Me',
+              '---',
+              'Here comes the content']),
+
+            (['',  # file starts with a blank line
+              '---',
+              'Todays motto: whatever'],
+             {},  # no context data
+             ['', '---', 'Todays motto: whatever']),  # everything preserved
+
+            (['This file has',
+              'no metadata whatsoever'],
+             {},
+             ['This file has', 'no metadata whatsoever']),
+
+        ]
+        plugin = PageContextPlugin()
+
+        for i, (lines, context, data) in enumerate(testdata):
+            self.assertEqual((context, data), 
+                             plugin.jekyll_style(lines),
+                             'Test Data No. %d failed' % i)
+
+        lines = ['---',
+                 'Author : Me',  # no end of metadata
+                 'Here comes the content']
+        with LogCapture() as l:
+            context, data = plugin.jekyll_style(lines)
+        self.assertEqual(len(l.records), 1)
+        self.assertEqual(l.records[0].levelname, 'WARNING')
+        self.assertEqual(l.records[0].msg, 'Page context data seem to end in line %d')
```

Comment:
When dealing with badly shaped input data, should Cactus issue a warning and then try its best or should it stop the build and make the user correct the data? I decided for the first way, but I'm absolutely open for different opinions.
(Think of having 100 input files generated by some other program …)